### PR TITLE
Reindex mint redeemers based on policy id order

### PR DESCRIPTION
**Pull request by errfrom**

### CODE REVIEW

File: src/Types/ScriptLookups.purs [hunk 1]
```diff
@@ -72,12 +72,13 @@ import Data.Lens.Record (prop)
...
 import Data.Map (Map, empty, fromFoldable, lookup, mapMaybe, singleton, union)
+import Data.Map (insert, toUnfoldable) as Map
 import Data.Maybe (Maybe(Just, Nothing), maybe)
 import Data.Newtype (class Newtype, over, unwrap, wrap)
 import Data.Show.Generic (genericShow)
 import Data.Symbol (SProxy(SProxy))
-import Data.Traversable (for, traverse)
-import Data.Tuple (Tuple(..), fst)
+import Data.Traversable (for, traverse, traverse_)
+import Data.Tuple (Tuple(..), fst, snd)
```
Comment (ngua) [inline src/Types/ScriptLookups.purs:81]: Explicit imports

File: src/Types/ScriptLookups.purs [hunk 2]
```diff
@@ -925,28 +944,24 @@ processConstraint mpsMap osMap = do
...
+          , index: zero
           , data: unwrap red
           , exUnits: mintExUnits
           }
+      -- Remove mint redeemers from array before reindexing.
+      _redeemers %= filter \(T.Redeemer { tag } /\ _) -> tag /= Mint
+      -- Reindex mint redeemers.
+      mintRedeemers <- lift $ reindexMintRedeemers mpsHash redeemer
+      -- Append reindexed mint redeemers to array.
+      _redeemers <>= map (\red -> red /\ Nothing) mintRedeemers
```
Comment (ngua) [inline src/Types/ScriptLookups.purs:963]:
```suggestion
      _redeemers <>= map (_ /\ Nothing) mintRedeemers
```

**Review by ngua — COMMENTED** [review]
These changes also need to be applied to master

File: src/Types/ScriptLookups.purs [hunk 3]
```diff
@@ -58,7 +58,7 @@ import Control.Monad.Reader.Trans (ReaderT)
 import Control.Monad.State.Trans (StateT, get, gets, put, runStateT)
 import Control.Monad.Trans.Class (lift)
 import Data.Array ((:), singleton, union) as Array
-import Data.Array (elemIndex, insert, toUnfoldable, zip)
+import Data.Array (elemIndex, filter, insert, mapWithIndex, toUnfoldable, zip)
```
Comment (vvtran) [inline src/Types/ScriptLookups.purs:61]: `elemIndex` redundant

**Pull request merged**